smarts/core/mission_planner.py

```diff
@@ -299,12 +300,28 @@ def cut_in_waypoints(self, sim, pose: Pose, vehicle, base_waypoint_generator):
     def uturn_waypoints(self, sim, pose: Pose, vehicle):
         # TODO: 1. Need to revisit the approach to calculate the U-Turn trajectory.
         #       2. Wrap this method in a helper.
+
+        ego_position = pose.position[:2]
+        ego_lane = self._road_network.nearest_lane(ego_position)
+        ego_wps = self._waypoints.waypoint_paths_on_lane_at(
+            ego_position, ego_lane.getID(), 60
+        )
+        if self._mission.task.initial_speed is None:
+            default_speed = ego_wps[0][0].speed_limit
```

Comment:
Is `ego_wps[0][0]` guaranteed to exist? Maybe it is helpful to set a variable here like 
`{description_of_ego_wps[0][0]} = ego_wps[0][0]`

Reply:
Thanks, ego_wps are coming from `waypoints`, where it calls for 
```
closest_linked_wp = self._closest_linked_wp_in_kd_tree_batched(
            [point], self._waypoints_by_lane_id[lane_id], lane_kd_tree, k=1
        )[0][0] 
```
which is the closest linked waypoint, at worst case scenario it will return the beginning or the end of the lane waypoint unless it is off the lanes where in that case the off road will be triggered.